science/commands/lift.py

```diff
@@ -139,26 +136,41 @@ def maybe_invert_lazy(file: File) -> File:
 
         for interpreter in application.interpreters:
             distribution = interpreter.provider.distribution(platform)
+            if distribution is None:
+                raise InputError(
+                    f"No compatible {providers.name(interpreter.provider)} distribution was found "
+                    f"for {platform}."
+                )
             if distribution:
                 distributions.append(distribution)
                 requested_files.append(maybe_invert_lazy(distribution.file))
         requested_files.extend(map(maybe_invert_lazy, application.files))
         if (actually_inverted := frozenset(inverted)) != lift_config.invert_lazy_ids:
             raise InputError(
                 "There following files were not present to invert laziness for: "
-                f"{', '.join(sorted(lift_config.invert_lazy_ids - actually_inverted))}"
+                f"{", ".join(sorted(lift_config.invert_lazy_ids - actually_inverted))}"
             )
 
-        if any(isinstance(file.source, Fetch) and file.source.lazy for file in requested_files):
-            ptex = a_scie.ptex(chroot, specification=application.ptex, platform=platform)
-            file_paths_by_id[ptex.id] = chroot / ptex.name
-            requested_files.appendleft(ptex)
-            argv1 = (
-                application.ptex.argv1
-                if application.ptex and application.ptex.argv1
-                else "{scie.lift}"
+        fetches_present = any(
+            isinstance(file.source, Fetch) and file.source.lazy for file in requested_files
+        )
+        if application.ptex or fetches_present:
```

Comment:
It turns out application.ptex is always non-null and so this change caused a `ptex` binary to be included in every scie, needed or not.

Reply:
Fixed in #124.